src/Approximations/overapproximate.jl

```diff
@@ -1510,3 +1510,121 @@ function overapproximate(am::AbstractAffineMap{N, <:AbstractHyperrectangle{N}},
     radius_MX = abs.(M) * radius_hyperrectangle(X)
     return Hyperrectangle(center_MXv, radius_MX)
 end
+
+"""
+    overapproximate(X::LazySet{N}, ::Type{<:Zonotope},
+                    dir::AbstractDirections{N};
+                    solver=default_lp_solver(N)) where {N<:Real}
+
+Overapproximate a polytopic set with a zonotope of minimal total generator sum
+using only generators in the given directions.
+
+### Input
+
+- `X`        -- polytopic set
+- `Zonotope` -- type for dispatch
+- `dir`      -- directions used for the generators
+- `solver`   -- (optional, default: `default_lp_solver(N)`) the backend used to
+                solve the linear program
+
+### Output
+
+A zonotope that overapproximates `X` and uses at most the directions provided in
+`dir` (redundant directions will be ignored).
+Under this constraint, the zonotope has the minimal sum of generator vectors.
+
+### Notes
+
+The algorithm only requires one representative of each generator direction and
+their additive inverse (e.g. only one of `[1, 0]` and `[-1, 0]`) and assumes
+that the directions are normalized.
+We preprocess the directions in that respect.
+
+### Algorithm
+
+We solve a linear program parametric in the vertices ``v_j`` of `X` and the
+directions ``d_k`` in `dir` presented in [1, Section 4.2] (adapting the notation
+to the one used in this library).
+
+```math
+    \\min \\sum_{k=1}^l c_k \\
+    s.t. \\
+    \\text{center} + \\sum_{k=1}^l b_{kj} * d_k = v_j \\quad \\forall j \\
+    -c_k ≤ b_{kj} ≤ c_k \\quad \\forall k, j \\
+    c_k ≥ 0 \\quad \\forall k, j \\
+```
+
+Note that the first type of side constraints is vector-based and that the
+nonnegativity constraints (last type) are not stated explicitly in [1].
+
+[1] Zonotopes as bounding volumes, L. J. Guibas et al, Proc. of Symposium on
+    Discrete Algorithms, pp. 803-812.
+"""
+# note: this set can generally be reduced
+function overapproximate(X::LazySet{N}, ::Type{<:Zonotope},
+                         dir::AbstractDirections{N};
+                         solver=default_lp_solver(N)) where {N<:Real}
+    function normalize_directions(dir::AD) where {AD}
+        dirs = Vector{Vector{N}}()
+        for d in dir
+            d_normalized = normalize(d)
+            if (d_normalized ∉ dirs) && (-d_normalized ∉ dirs)
+                push!(dirs, d_normalized)
+            end
+        end
+        return dirs
+    end
+
+    n = dim(X)
+    dirs = normalize_directions(dir)
```

Comment:
this normalization should be optional, thinking that as use case for this function is to fix a template, normalize it (outside the function) and call `overapproximate` many times.

Reply:
Currently there is no implementation satisfying the normalization. We can do that later.

Reply:
you can at least save the step `d_normalized = normalize(d)`

Reply:
Only for box directions and this is a cheap operation so I would prefer to not complicate the code with conditions for one type.

Reply:
> Only for box directions

why? polar and spherical directions also have (2-norm) one. and custom directions may have it if one defines them.

> complicate the code with conditions for one type

hmm? we can dispatch on the directions type / the trait.

Reply:
I added a TODO in the code.